distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java

```diff
@@ -0,0 +1,205 @@
+     */
+    public void stop() {
+        cswStubServer.stop();
+    }
+
+    /**
+     * Resets the Restito server by stopping and starting it.
+     */
+    public void reset() {
+        // There is no way in version 0.7 of Restito to clear the calls and reset the
+        // StubServer so we need to stop the current one and create a new instance.
+        // TODO - Replace this code when the new version of Restito is used.
+        stop();
+        start();
+    }
+
+    /**
+     * Returns StubHttp's whenHttp method with the stub server as the parameter. Use is the same.
+     * Sets up an Action when a specific Http call is made.
+     *
+     * @return StubHttp using whenHttp method with the stub server as the parameter.
+     */
+    public StubHttp whenHttp() {
+        return StubHttp.whenHttp(cswStubServer);
+    }
+
+    /**
+     * Returns VerifyHttp's verifyHttp method with the stub server as the parameter. Use is the same.
+     * Provides itest verification when a specific Http call is made.
+     *
+     * @return VerifyHttp using verifyHttp method with the tub server as the parameter.
+     */
+    public VerifyHttp verifyHttp() {
+        return VerifyHttp.verifyHttp(cswStubServer);
+    }
+
+    /**
+     * Get the post being used by the stub server.
+     *
+     * @return the port being used by the stub server.
+     */
+    public int getPort() {
+        return port;
+    }
+
+    /**
+     * Set a specific capability Response. Overwrites default/previous response condition.
+     *
+     * @param resourceName name of capability file
+     * @throws IOException
+     */
+    private void setCapabilityResponse(String resourceName) throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream(
+                "/" + resourceName)) {
+            String document = substituteTags(IOUtils.toString(inputStream));
+
+            LOGGER.debug("Capability response: \n{}", document);
+
+            whenHttp().match(get("/services/csw"))
+                    .then(ok(), contentType("text/xml"), bytesContent(document.getBytes()));
+        }
+    }
+
+    /**
+     * Set a specific query Response. Overwrites default/previous response condition.
+     *
+     * @param resourceName name of resource file
+     * @throws IOException
+     */
+    public void setQueryResponse(String resourceName) throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream("/" + resourceName)) {
+            String document = substituteTags(IOUtils.toString(inputStream));
+
+            LOGGER.debug("Query response: \n{}", document);
+
+            whenHttp().match(post("/services/csw"), withPostBodyContaining("GetRecords"))
+                    .then(ok(), contentType("text/xml"), bytesContent(document.getBytes()));
+        }
+    }
+
+    private void setDefaultCapabilityResponse() throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream(
```

Comment:
Unfortunately DynamicUrl is in AbstractIntegrationTest, which is in test-itest-ddf.